src/v/cluster/topics_frontend.cc

```diff
@@ -841,6 +843,27 @@ ss::future<topic_result> topics_frontend::do_create_topic(
     if (!units) {
         co_return make_error_result(assignable_config.cfg.tp_ns, units.error());
     }
+
+    // Enforce cluster-wide topic count limit only on user created topics.
+    auto max_user_topics_opt = _max_user_topics();
+    if (max_user_topics_opt.has_value()) {
```

Comment:
You can put the previous line inside the if, which saves a line and keeps the local to its minimal scope, like:

```
if (auto max_user_topics_opt = _max_user_topics()) {
```

Reply:
Nice, forgot about this. Will switch over.